core/eds.go

```diff
@@ -17,20 +22,45 @@ import (
 // extendBlock extends the given block data, returning the resulting
 // ExtendedDataSquare (EDS). If there are no transactions in the block,
 // nil is returned in place of the eds.
-func extendBlock(data types.Data, appVersion uint64) (*rsmt2d.ExtendedDataSquare, error) {
+func extendBlock(data types.Data, appVersion uint64, fn rsmt2d.TreeConstructorFn) (*rsmt2d.ExtendedDataSquare, error) {
```

Comment:
Reworked to pass only options. 
Latere, we would want rsmt2d.ExtendedDataSquare to optionally contain tree nodes. This will clean up whole context passing mess and will create a much cleaner api.